### DISCUSSION TITLE
Public Variables: error on deploy ["Environment Variable not found for file"]

### DISCUSSION THREAD

**Discussion by vguccione01** [mia-platform/community#305] [Q&A] [answered]
### Which section?

Console

### Which is the version?

1.87.1

### Description

When I try to deploy a service using a public variable defined in the project I get the following error on gitlab pipeline

```
WARN: can't read input file at path /builds/clients/**omit**/**my-project-name**/Configurations/configuration/development
Environment Variable MY_PUBLIC_VARIABLE not found for file: apiVersion: apps/v1
```

everything works fine using variables defined into /infrastructure/variables section

### Actual Outcome

Deploy failed with error above

### Expected Outcome

Successful deploy and variable correctly interpolated into service

**Comment by fredmaggiowski** [comment]
Hi @vguccione01 thanks for reaching out!

Have you configured `MY_PUBLIC_VARIABLE` within the *Public Variables* section in the Design Area with a specific value for the environment you are deploying to?

**Reply by vguccione01** [reply]
Hi @fredmaggiowski !
Yes, I configured `MY_PUBLIC_VARIABLE` into the _Public Variables_ section. My only environment is `development`.
Then I interpolated the value of the variable using the `{{ }}` notation into the _Environment Variable_ section of my microservice 

------------------------
![Screenshot 2023-09-04 alle 17 31 29](https://github.com/mia-platform/community/assets/114938987/e84dbc2a-6248-4c1b-8528-83e672428a65)
------------------------
![Screenshot 2023-09-04 alle 17 37 23](https://github.com/mia-platform/community/assets/114938987/3403d5e9-de83-4306-b980-7c36c328c520)

**Reply by fredmaggiowski** [reply]
OK, thanks for the info; can you please verify that you have one of these files:

 - `overlays/development/variables.env` if your project is using Kustomize
 - `variables/development.env` if your project is not using Kustomize

One of the two must exists and have a content like 

```env
MY_PUBLIC_VARIABLE=the value you wrote in console
```

NB: the key may have an additional `MIA_` prefix depending on your project configuration but it's not something to be bothered about, let's begin with verifying the file exists

**Reply by vguccione01** [reply]
I just checked. Both files exists:
- `overlays/development/variables.env` contains `MY_PUBLIC_VARIABLE` and its value
- `variables/development.env` is an empty file

**Reply by Noste92** [reply, marked as answer]
Hi @vguccione01, @fredmaggiowski ,

I'm working on the same project as @vguccione01. We have both environments files so it seems that someone changed accidentally the property ```"projectStructure": "kustomize"``` in the deploy section from the CMS. 
By removing that property everything works.

Thank you

**Reply by vguccione01** [reply]
Thank you @Noste92 
I confirm that now everything works perfectly!
